rhoso-observability.md

```diff
@@ -0,0 +1,31 @@
+# Openstack Observability
+
+Protocols and semantic conventions for metrics in Red Hat OpenStack Services on Openshift (OSP 18+). This document currently covers the metrics portion of observability.
+
+## Ingestion protocols
+
+All metrics in OpenStack are collected via HTTP scrape endpoints that present the metrics in the [Prometheus Exposition Format](https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md)
+
+## Data Model
+
+Metrics in OpenStack conform to the [Prometheus Data Model](https://prometheus.io/docs/concepts/data_model/)
+
+## Semantic Conventions
+
+The semantics of metrics data in OpenStack are inherited from the underlying collector agents in use.
+
+* For VMs, the agent is Ceilometer, and the semantics are partially described in the [Ceilometer Measurements documentation](https://docs.openstack.org/ceilometer/latest/admin/telemetry-measurements.html)
+  * NOTE: Only a subset of "pollsters" are enabled.
+  * [Ceilometer Compute Agent Polling config](https://github.com/openstack-k8s-operators/telemetry-operator/blob/main/templates/ceilometercompute/config/polling.yaml#L5)
+  * [Ceilometer Central Agent Polling config](https://github.com/openstack-k8s-operators/telemetry-operator/blob/main/templates/ceilometercentral/config/polling.yaml.j2#L5)
+* For compute node Hosts, the agent is node_exporter, and the semantics are partially described in the [Prometheus Metric and Label Naming documentation](https://prometheus.io/docs/practices/naming/). 
+  * NOTE: Only a subset of [all "Collectors"](https://github.com/prometheus/node_exporter?tab=readme-ov-file#collectors) are enabled.
+  * [Node Exporter Collector config](https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/roles/edpm_telemetry/tasks/install.yml#L50)
+
+A thorough inventory of all labels in use and their specific semantics is still a WIP.
+
+## USE/RED and (future) OTEL Compatibility
+
+The following table summarizes the core metrics for VMs and compute node Hosts. Not all collected metrics are represented, but it covers everything that goes into our stock dashboards and represents the current (Apr 2024) best-fit to the USE model.
+
```

Comment:
This is an export/hack-up from my google sheet. Not 100% happy with the result and it will be a PITA to update, but consider it representative for now.

Reply:
It renders well here. I would suggest adding link to the source google sheet as a comment. Even though it is private.

Reply:
I tried to make it public (I'm not shy) but I'm not sure it's even possible. The link is here: 

https://docs.google.com/spreadsheets/d/1oz8uSR5T51ocMUUu_z9mvr3S3Lsbn71mh42KWef_FBk/edit?usp=sharing

Do you think it's worth adding to the actual text? I figured this stuff is supposed to be customer-facing and I don't want to add a dead link